src/CallWebhookJob.php

```diff
@@ -70,13 +73,18 @@ public function handle()
                 'timeout' => $this->requestTimeout,
                 'verify' => $this->verifySsl,
                 'headers' => $this->headers,
-            ], $body));
+                'on_stats' => function (TransferStats $stats) {
 
-            if (! Str::startsWith($this->response->getStatusCode(), 2)) {
-                throw new Exception('Webhook call failed');
-            }
+                    $response = $stats->getResponse();
+                    $this->stats = $stats;
 
-            $this->dispatchEvent(WebhookCallSucceededEvent::class);
+                    if (! Str::startsWith($response->getStatusCode(), 2)) {
```

Comment:
Why do we throw the exception in on_stats? The previous code looks cleaner in this regard. Imho we just get the stats and nothing more here.

Reply:
I was originally under the impression that stats would not get set since the `on_stats` hook is called after the responsee 
 and the event would be dispatched without the stats var set. Anyway, I've reverted and it works the same. I'll be adding test ASAP and updating the branch.